actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -470,7 +480,17 @@ def initialize(parent_jar)
 
       def [](name)
         if data = @parent_jar[name.to_s]
-          parse name, data
+          if request.use_cookies_with_metadata
+            value = parse name, data, purpose: "cookie.#{name}"
+
+            if value.nil? && request.honor_cookies_without_metadata
```

Comment:
Use cases for the new `action_dispatch.honor_cookies_without_metadata` config are:
1. Can be used to reject old cookies without the new metadata or honor them, while `use_cookies_with_metadata` is true i.e. we're now writing cookies with the new metadata embedded and we're still able to read old cookies without them. Nevertheless, without the  `action_dispatch.honor_cookies_without_metadata` config we would not be able to control for how long we want to accept the old cookies, we might have accepted them forever.

2. Let's imagine if the config was absent at line 486, then in case of a new rails app which doesn't have any old cookies without the embedded metadata out in the wild, we would be parsing through twice and since the config defaults to false we'll only parse once.